assignments/assignment01.rb

```diff
@@ -68,13 +80,168 @@ def my_reverse(s)
 # Write a method `palindrome?`
 # that determines whether a string is a palindrome
 def palindrome?(s)
-  # your implementation here
+  g = s.gsub(/\W+/, "")
+  f = g.downcase
+  if f.reverse == f
+    puts "true"
+  else
+    puts "false"
+  end
 end
 
 # Your method should generate the following results:
-palindrome? "abba"                             #=> true
-palindrome? "aBbA"                             #=> true
-palindrome? "abb"                              #=> false
+palindrome?("abba")                             #=> true
+palindrome?("aBbA")                             #=> true
+palindrome?("abb")                             #=> false
+
+palindrome?("Able was I ere I saw elba")       #=> true
+palindrome?("A man, a plan, a canal, Panama")   #=> true
+puts "---END OF PROBLEM 3---"
+
+#----------------------------------------------------------------
+
+puts "---START OF HOMEWORK 2---"
+puts ""
+
+def to_sentence(array)
+  if array.length == 1
+    puts array
+  elsif array.length > 1
+    last_element = []
+    last_element << array.pop
+    p = array.flatten.join(", ")
+    puts p+" and "+last_element[0].to_s
+  end
+end
+
+to_sentence([])
+to_sentence(["john"])
+to_sentence(["john", "paul"])
+to_sentence([1, "paul", 3, "ringo"])
+puts "---END OF PROBLEM 1---"
+
+def mean(array)
+  sum = 0
+  array.each {|i| sum += i}
+  the_mean = sum/array.length.to_i
+  puts the_mean.to_f
+end
+
+def median(array)
+  ordered_array = array.sort
+  number_of_elements = array.length
+  if array.length % 2 != 0
+    median_pos = number_of_elements.to_i/2
+    puts ordered_array[median_pos].to_f
+  else array.length % 2 == 0
+    median1 = ordered_array[array.length.to_i/2]
+    median2 = ordered_array[array.length.to_i/2-1]
+    puts (median1.to_f+median2.to_f)/2
+  end
+end
+
+mean([1, 2, 3])
+mean([1, 1, 4])
+puts ""
+
+median([1, 2, 3])
+median([1, 1, 4])
+puts "---END OF PROBLEM 2---"
+
+def pluck(hash, target)
```

Comment:
I think you misunderstood this problem. The first argument is an array. The second argument is an arbitrary key. The solution is the single line:
ary.map {|item| item[key]}